src/main/java/exam2/lotto/PurchaseTicket.java

```diff
@@ -0,0 +1,105 @@
+package exam2.lotto;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * 구입한 로또 클래스
+ */
+public class PurchaseTicket {
+
+    private List<LottoNumbers> lottoNumbersList = new ArrayList<>();
+    private int count = 0;
+
+    private Map<LottoPrize, Integer> matchCount = new HashMap<>();
+    private double rate;
+
+    public PurchaseTicket(List<LottoNumbers> lottoNumbersList) {
+        this.lottoNumbersList = lottoNumbersList;
+        this.count = lottoNumbersList.size();
+        initMatchCount();
+    }
+
+    public static PurchaseTicket createAutoLottoNumber(int price) {
+        if (price < LottoPrice.PRICE_1000.getPrice()) {
+            throw new IllegalArgumentException("로또 구입 가격은 1000원 이상 이여야 합니다");
+        }
+
+        int count = price / LottoPrice.PRICE_1000.getPrice();
+
+        List<LottoNumbers> lottoNumbersList = new ArrayList<>();
+
+        for (int i = 0; i < count; i++) {
+            lottoNumbersList.add(new LottoNumbers());
+        }
+
+        return new PurchaseTicket(lottoNumbersList);
+    }
+
+    /**
+     * 모든 로또 번호의 당첨 개수를 계산 합니다
+     *
+     * @param winning
+     */
+    public void calculateMatchCount(WinningNumber winning) {
+        for (int i = 0; i < lottoNumbersList.size(); i++) {
+
+            int count = (int) lottoNumbersList.get(i).getNumberList().stream()
+                .filter(lottoNum -> winning.getLottoNumbers().getNumberList().contains(lottoNum))
+                .count();
+
+            this.matchCount.put(LottoPrize.getLottoPrize(count),
+                this.matchCount.get(LottoPrize.getLottoPrize(count)) + 1);
+        }
```

Comment:
```suggestion
            for (LottoNumbers lottoNumbers : lottoNumbersList) {
                  int count = (int) lottoNumbers.getNumberList().stream()
                    .filter(lottoNum -> winning.getLottoNumbers().getNumberList().contains(lottoNum))
                    .count();
                   
                  // 한 템포 쉬어가기
                  LottoPrize lottoPrize = LottoPrize.getLottoPrize(count);
      
                  this.matchCount.put(lottoPrize, this.matchCount.get(lottoPrize) + 1);
            }
```
for i > for each문으로 변경하면
다음과 같이 작성해볼수도 있을꺼 같아요.

그리고, 필터부분의 LottoNumbers클래스에서 해당 로또번호가 있는지 없는지 판별하는 메서드가 있으면 조금더 간결하게 표현이 될꺼 같아요